### ISSUE TITLE
Illegal onEvent method - ProGuard

### ISSUE THREAD

**Issue by DavidEdwards** [greenrobot/EventBus#133]
I am receiving this error:

Illegal onEvent method, check for typos

The error appears to be caused by a method name changing in my compile process by ProGuard. Here is a sample of the log I created to test:

> methodName=onEvent
> modifierString=
> methodName=onEvent$3884d48a
> modifierString=$3884d48a
> skipMethodVerificationForClasses={}

The solution is to check for the $ character at the start of the modifierString.

This diff solves the issue:

```
@@ -91,7 +91,7 @@ class SubscriberMethodFinder {
                             } else if (modifierString.equals("Async")) {
                                 threadMode = ThreadMode.Async;
                             } else {
-                                if (skipMethodVerificationForClasses.containsKey(clazz)) {
+                                if (skipMethodVerificationForClasses.containsKey(clazz) || modifierString.startsWith("$")) {
                                     continue;
                                 } else {
                                     throw new EventBusException("Illegal onEvent method, check for typos: " + method);
```

**Comment by bluef** [comment]
We met the same issue, and we searched the mapping.txt and found one line like this:

```
448:462:void onEvent$6f60fb76(android.support.v7.widget.RecyclerView$SmoothScroller$Action) -> onEvent$6f60fb76
```

RecyclerView introduce a method named onEvent$6f60fb76 which breaks SubscriberMethodFinder.findSubscriberMethods. @DavidEdwards 's way is a good way to fix it.

**Comment by sdeff** [comment]
I had a similar issue and I contacted the creator of ProGuard/DexGuard and he sent me a 'workaround' for this.

Just add:
-keepclassmembers,includedescriptorclasses class ** { public void onEvent*(**); }

To your config file to prevent ProGuard/DexGuard's optimization step to add a suffix to the method name.

**Comment by jayzhou215** [comment]
no, you cannot fix this issus by this. you just hide the bug, the method onEvent$() won't execute in logic if fix by this!

**Comment by jayzhou215** [comment]
@sdeff Your answer should be the right solution.

**Comment by cs-victor-nascimento** [comment]
Stumbled upon this recently. This should be in the wiki of the project.
